### DISCUSSION TITLE
ProcessTimedOutException after upgrading to puppeteer 19.7.0

### DISCUSSION THREAD

**Discussion by nbyloff** [spatie/browsershot#752] [Q&A]
I have been using this package with puppeteer v14.4.1 for a while now in a docker container. I have noticed I am having font loading issues and the only solution that worked was base64 the font in my CSS file, but that made the PDF size explode from 3MB to 45MB.

Anyways, I am trying to upgrade to a more current version to see if I can get my fonts to load without the file size issue. I have upgraded to puppeteer v.19.7.0. ( I installed node 18.x inside the container). I am testing on a Mac M1 machine, but I run my docker container with `--platform linux/amd64`. Within the docker container, I make sure the correct chromium is installed with `node node_modules/puppeteer/install.js`. 

Based on the install path of chromium, I update my Browsershot PDF command: 

````

    $pdf = Browsershot::html(View::make($view, $data))
      ->setChromePath('/home/web/.cache/puppeteer/chrome/linux-1095492/chrome-linux/chrome')
      ->timeout(300)
      ->addChromiumArguments([
        'font-render-hinting' => 'none',
        'force-color-profile' => 'srgb',
        'headless'=> true
      ])
      ->noSandbox()
      ->landscape()
      ->userAgent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36')
      ->setOption('headless', 'new')
      ->setOption('args', [
        '--disable-web-security',   // maybe fixes issues related to cors, etc
        '--disable-gpu',
        '--disable-setuid-sandbox',
        '--disable-dev-shm-usage' 
      ])
      ->waitUntilNetworkIdle()
      ->showBackground();
````

The problem is, it hangs indefintely until it hits my timeout. On v14.4.1 this process takes less than 60 seconds so I know it's not still running. It's not giving me any additional information other than there is a timeout. Do I have any options to debug this process? I am wondering if puppeteer is spitting out any information the ProcessTimeOutException doesn't know about. If I run `ps` inside the container, I do see a chrome executable running, but not high usage when it's working normally. Just like 12% vs. 103% CPU usage.

Currently I have stripped my HTML down to a single page that is not loading any external resources like JS or CSS files. Just pure HTML, and it still hangs. 

$ npm --version
9.5.1
$ node --version
v18.16.1

**Comment by benddailey** [comment]
@nbyloff Did you come up with any solution to this? I believe I am running into a similar issue on a new dev machine I just setup. Thanks!

**Comment by nbyloff** [comment]
@benddailey No and I haven't had time to troubleshoot a potential solution. [This user on SO](https://stackoverflow.com/a/76191625/239375) solved a different solution with a chromium image.

I was going to test if [using this image](https://hub.docker.com/r/zenika/alpine-chrome ) with the correct version solved my issues. At worst, it will allow me to eliminate Chromium version being the reason it's timing out. I'll eventually test and report back, but if it works for you, please let me know.
